hyperstone/plugins/hooks/context.py

```diff
@@ -0,0 +1,23 @@
+from typing import Optional, TYPE_CHECKING
+
+from hyperstone.emulator import HyperEmu
+
+if TYPE_CHECKING:
+    from hyperstone.plugins.hooks.base import ActiveHook
+
+
+class Context:
```

Comment:
Why do we need two classes? can we just inherit from dict here?

Reply:
DictContext is user facing and is present as a good alternative to the old context system. However by default we don't want to support a dict based context as it is untyped and implicit. As such we supply an alternative but don't make it the default implementation.